backend/pkg/api/handle_topics_integration_test.go

```diff
@@ -0,0 +1,406 @@
+	testutil.CreateTestData(t, context.Background(), s.kafkaClient, s.kafkaAdminClient,
+		testutil.TopicNameForTest("get_topics_2"))
+
+	defer func() {
+		s.kafkaAdminClient.DeleteTopics(context.Background(),
+			testutil.TopicNameForTest("get_topics_0"),
+			testutil.TopicNameForTest("get_topics_1"),
+			testutil.TopicNameForTest("get_topics_2"))
+	}()
+
+	t.Run("happy path", func(t *testing.T) {
+		ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+		defer cancel()
+
+		res, body := s.apiRequest(ctx, http.MethodGet, "/api/topics", nil)
+
+		assert.Equal(200, res.StatusCode)
+
+		type response struct {
+			Topics []*console.TopicSummary `json:"topics"`
+		}
+
+		getRes := response{}
+
+		err := json.Unmarshal(body, &getRes)
+		assert.NoError(err)
+
+		assert.Len(getRes.Topics, 3)
+		assert.Equal(testutil.TopicNameForTest("get_topics_0"), getRes.Topics[0].TopicName)
+		assert.Equal(testutil.TopicNameForTest("get_topics_1"), getRes.Topics[1].TopicName)
+		assert.Equal(testutil.TopicNameForTest("get_topics_2"), getRes.Topics[2].TopicName)
+	})
+
+	t.Run("no see permission", func(t *testing.T) {
+		topicName := testutil.TopicNameForTest("get_topics_1")
+
+		oldHooks := s.api.Hooks
+		newHooks := newAssertHooks(t, map[string]map[string]assertCallReturnValue{
+			"CanSeeTopic": {
+				testutil.TopicNameForTest("get_topics_0"): assertCallReturnValue{BoolValue: true, Err: nil},
+				topicName: assertCallReturnValue{BoolValue: false, Err: nil},
+				testutil.TopicNameForTest("get_topics_2"): assertCallReturnValue{BoolValue: true, Err: nil},
+			},
+			"AllowedTopicActions": {
+				"any": assertCallReturnValue{BoolValue: true, Err: nil},
+			},
+		})
+
+		if newHooks != nil {
+			s.api.Hooks = newHooks
+		}
+
+		defer func() {
+			if oldHooks != nil {
+				s.api.Hooks = oldHooks
+			}
+		}()
+
+		ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+		defer cancel()
+
+		res, body := s.apiRequest(ctx, http.MethodGet, "/api/topics", nil)
+
+		assert.Equal(200, res.StatusCode)
+
+		type response struct {
+			Topics []*console.TopicSummary `json:"topics"`
+		}
+
+		getRes := response{}
+
+		err := json.Unmarshal(body, &getRes)
+		assert.NoError(err)
+
+		assert.Len(getRes.Topics, 2)
+		assert.Equal(testutil.TopicNameForTest("get_topics_0"), getRes.Topics[0].TopicName)
+		assert.Equal(testutil.TopicNameForTest("get_topics_2"), getRes.Topics[1].TopicName)
+	})
+
+	t.Run("allow topic action error", func(t *testing.T) {
+		topicName := testutil.TopicNameForTest("get_topics_1")
+
+		oldHooks := s.api.Hooks
+		newHooks := newAssertHooks(t, map[string]map[string]assertCallReturnValue{
+			"CanSeeTopic": {
+				"any": assertCallReturnValue{BoolValue: true, Err: nil},
+			},
+			"AllowedTopicActions": {
+				testutil.TopicNameForTest("get_topics_0"): assertCallReturnValue{SliceValue: []string{}, Err: nil},
+				topicName: assertCallReturnValue{Err: &rest.Error{
+					Err:     fmt.Errorf("error from test"),
+					Status:  505,
```

Comment:
Do we return status 505 somewhere? What is this?

What do you think about using the `http.StatusCode` consts instead of the status code number?

Reply:
No normally this would return Unauthorized usually. I was using a different signal value since it's a mocked failure. But I have changed it to return Unauthorized to represent a real world scenario.